### PR TITLE
test: add threshold e2e tests

### PR DESCRIPTION
**Pull request by jamesgorrie**
Adds some tests checking that we display the right product that will get sent to the `support-workers` based on the thresholds.

Tests include `promoCodes` too (we'd recently seen some bugs come through with this fixed in https://github.com/guardian/support-frontend/pull/5908, https://github.com/guardian/support-frontend/pull/5904, https://github.com/guardian/support-frontend/pull/5906, https://github.com/guardian/support-frontend/pull/5903).

### CODE REVIEW

File: support-e2e/tests/tieredCheckout.test.ts [hunk 1]
```diff
@@ -182,3 +182,68 @@ test.describe('Subscribe (S+) incl PromoCode via the Tiered checkout', () => {
...
+
+			const orderSummaryHeading = page.getByRole('heading', {
+				name: 'Your subscription',
+			});
+			const orderSummarySection = page
+				.locator('section')
+				.filter({ has: orderSummaryHeading })
+				.nth(1);
+
+			expect(
```
Comment (paul-daniel-dempsey) [inline support-e2e/tests/tieredCheckout.test.ts:244]: Should we be checking the `total` amount is as expected too? Tests 5/8 look wrong on laywright but perhaps the total fix from previous playwright tests needs to go in first?

Comment (paul-daniel-dempsey) [reply support-e2e/tests/tieredCheckout.test.ts:240]: fyi: updated and tests 5/8 are fine

**Review by paul-daniel-dempsey — APPROVED** [review]
Looks good, see comment on total check too?

File: support-e2e/tests/tieredCheckout.test.ts (hunk 1 above)
Comment (jamesgorrie) [reply support-e2e/tests/tieredCheckout.test.ts:240]: I don't really want to be checking the total here (this is done with the promoCode and checkout tests. 

This is just saying "Am I showing you the right product given our thresholds".

Comment (jamesgorrie) [reply support-e2e/tests/tieredCheckout.test.ts:240]: You're right - the branch needed updating with `main`.

**Pull request merged**